enterprise_access/apps/provisioning/models.py

```diff
@@ -416,18 +418,52 @@ def process_input(self, accumulated_output=None, **kwargs):
         else:
             catalog_uuid = str(accumulated_output.create_catalog_output.uuid)
 
-        result_dict = get_or_create_subscription_plan(
-            customer_agreement_uuid=str(accumulated_output.create_customer_agreement_output.uuid),
-            existing_subscription_list=accumulated_output.create_customer_agreement_output.subscriptions,
-            plan_title=self.input_object.title,
-            catalog_uuid=catalog_uuid,
-            opp_line_item=self.input_object.salesforce_opportunity_line_item,
-            start_date=self.input_object.start_date.isoformat(),
-            expiration_date=self.input_object.expiration_date.isoformat(),
-            desired_num_licenses=self.input_object.desired_num_licenses,
-            product_id=self.input_object.product_id,
-        )
-        return self.output_class.from_dict(result_dict)
+        customer_agreement_uuid = str(accumulated_output.create_customer_agreement_output.uuid)
+
+        try:
+            result_dict = get_or_create_subscription_plan(
+                customer_agreement_uuid=customer_agreement_uuid,
+                existing_subscription_list=accumulated_output.create_customer_agreement_output.subscriptions,
+                plan_title=self.input_object.title,
+                catalog_uuid=catalog_uuid,
+                opp_line_item=self.input_object.salesforce_opportunity_line_item,
+                start_date=self.input_object.start_date.isoformat(),
+                expiration_date=self.input_object.expiration_date.isoformat(),
+                desired_num_licenses=self.input_object.desired_num_licenses,
+                product_id=self.input_object.product_id,
+            )
+            self.synchronize_checkout_intent()
+            return self.output_class.from_dict(result_dict)
+        except Exception as exc:
+            self.synchronize_checkout_intent(exc)
+            raise GetCreateSubscriptionPlanException(
+                f'Failed to get/create subscription plan for customer agreement uuid {customer_agreement_uuid}'
+            ) from exc
+
+    def synchronize_checkout_intent(self, exc=None):
+        """
+        Links this step's workflow to the related CheckoutIntent, if any.
+        If found, also updates the CheckoutIntent's state.
+        """
+        workflow = self.get_workflow_record()
+        enterprise_slug = workflow.input_object.create_customer_input.slug
+        checkout_intent = CheckoutIntent.filter_by_name_and_slug(
+            slug=enterprise_slug,
+        ).filter(
+            state=CheckoutIntentState.PAID,
```

Comment:
I worry this will get in the way of recovering from ERRORED_PROVISIONING on attempting to retry.

Reply:
oooh good call. updating